File: e-livraria-web/src/models/TradeDevolutionCoupon.ts
import { Client } from "./Client";

export interface ITradeDevolutionCoupon{
    id?: number;
    name?: string;
    value?: number;
    client?: Client
    used?: boolean;
}

export class TradeDevolutionCoupon{
    id: number;
    name: string;
    value: number;
    client: Client;
    used: boolean;

    constructor( {
        id = 0,
        name = '',
        value = 0,
        client = new Client({}),
        used = false
    }: ITradeDevolutionCoupon){
        this.id = id;
        this.name = name;
        this.value = value;
        this.client = client;
        this.used = used;
    }

    // get id(){
    //     return this._id;
    // }
    // set id(id: number){
    //     this._id = id;
    // }

    // get name(){
    //     return this._name;
    // }
    // set name(name: string){
    //     this._name = name;
    // }

    // get value(){
    //     return this._value;
    // }
    // set value(value: number){
    //     this._value = value;
    // }
}

File: ecommerceLivraria/src/main/java/com/nrabello/ecommerceLivraria/model/TradeDevolutionCoupon.java
package com.nrabello.ecommerceLivraria.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "Cupons_Troca_Devolucao")
public class TradeDevolutionCoupon {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name="ctd_id")
    private Long id;

    @Column(name="ctd_nome")
    private String name;

    @Column(name="ctd_valor")
    private Double value;

    @Column(name="ctd_usado")
    private Boolean used;

    @ManyToMany(mappedBy = "tradeDevolutionCoupons", fetch = FetchType.LAZY)
    @JsonIgnore
    private Set<Order> orders = new HashSet<>();

    public TradeDevolutionCoupon(){

    }
    public TradeDevolutionCoupon(Long id, String name, Double value, Boolean used ,Set<Order> orders) {
        this.setId(id);
        this.setName(name);
        this.setValue(value);
        this.setUsed(used);
        this.setOrders(orders);
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Double getValue() {
        return value;
    }

    public void setValue(Double value) {
        this.value = value;
    }

    public Boolean getUsed() {
        return used;
    }

    public void setUsed(Boolean used) {
        this.used = used;
    }

    public Set<Order> getOrders() {
        return orders;
    }
    public void setOrders(Set<Order> orders) {
        this.orders = orders;
    }
}
